R/print.data.table.R

```diff
@@ -15,6 +15,8 @@ print.data.table = function(x, topn=getOption("datatable.print.topn"),
   # trunc.cols - should only the columns be printed that can fit in the console? (FALSE)
   if (!col.names %chin% c("auto", "top", "none"))
     stop("Valid options for col.names are 'auto', 'top', and 'none'")
+  if (!is.logical(trunc.cols) | is.na(trunc.cols))
```

Comment:
operator `|` will not do a good job, it is a vectorized "or", and because is.logial and is.na are vectorized as well, we may ended up passing a logical vector to `if`.
because we want this to be scalar, the safe approach can be testing for `isTRUE || isFALSE`

Reply:
Fixed.